xdsl/backend/csl/print_csl.py

```diff
@@ -352,7 +373,13 @@ def print_block(self, body: Block):
                     with self.descend(loop_definition) as inner:
                         inner.print_block(bdy.block)
                 case scf.Yield():
-                    pass
+                    if (
+                        isinstance(op.parent, Block)
+                        and isinstance(op.parent.parent, Region)
+                        and isinstance(ifop := op.parent.parent.parent, scf.If)
+                    ):
```

Comment:
Can be replaced with `isinstance(op.parent_op(), scf.If)`

Reply:
@dk949 does printing of `scf.Yield` slightly differently for `scf.For` loops - basically, instead of passing a value to the `scf.Yield` op, it is assigned to the outside var directly. I'm matching this for now. Happy to hear your thoughts on this?

Reply:
I don't have any opinion on it, it should result in the same code anyway, right?